### ISSUE TITLE
Refactor GraphEvent

### ISSUE THREAD

**Issue by jsbean** [jsbean/DNM#215]
```Swift
class GraphEvent: CALayer {

    // model
    private let leaf: DurationNode

    // position
    private let x: CGFloat
    private let width: CGFloat

    // orientation
    private let stemDirection: StemDirection

    // graphical representations of ArticulationTypes
    private var articulations: [Articulation] = []

    // this is currently not a calculated property, and is done after articulations are positioned
    public var slurConnectionY: CGFloat { ... }

    public var maxInfoY: CGFloat { ... }
    public var minInfoY: CGFloat { ... }
    public var maxY: CGFloat { ... }
    public var minY: CGFloat { ... }
    public var stemEndY: CGFloat { ... }

    public init(
         leaf: DurationNode, 
         graph: GraphLayer, 
         x: CGFloat, 
         width: CGFloat, 
         stemDirection: StemDirection
    )
    {
         ...
    }
}
```

**Comment by jsbean** [comment]
Clarify order of operations in `.build()`

**Comment by jsbean** [comment]
Refactor `StaffEvent`

**Comment by jsbean** [comment]
Refactor `InstrumentEvent`

```Swift
struct InstrumentEvent {
    
    let x: CGFloat
    var graphEvents: [GraphEvent] = []

    ...
}
```

**Comment by jsbean** [comment]
Implement `InstrumentEventRest`

**Comment by jsbean** [comment]
Render `GraphLines` with `GraphEvents`, as opposed through some reach around that is going on now. Now that they should know `x` and `width` values.

**Comment by jsbean** [comment]
Closed with #221
